libbeat/docs/breaking.asciidoc

```diff
@@ -2,15 +2,148 @@
 == Breaking changes
 
 As a general rule, we strive to keep backwards compatibility between minor
-versions (e.g.  5.x to 5.y) so you can upgrade without any configuration file
+versions (e.g.  6.x to 6.y) so you can upgrade without any configuration file
 changes, but there are breaking changes between major versions (e.g. 5.x to
 6.y).
 
+* <<breaking-changes-6.0>>
 * <<breaking-changes-5.0>>
 
 We also list the breaking changes at the top of the <<release-notes,release notes>>
 for each version.
 
+[[breaking-changes-6.0]]
+=== Breaking changes in 6.0
+
+This section discusses the main changes that you should be aware of if you
+upgrade the Beats from version 5.x to 6.x. Please also review the relevant
+Breaking Changes sections of the <<release-notes,release notes>>.
+
+// TODO: better link to the consolidated release notes for 6.0.0.
+
+
+[[breaking-changes-spooler-removed]]
+==== Filebeat spooler removed
+
+Version 6.0 comes with a new architecture for the internal pipeline of all
+Beats. This architecture refactoring is mostly internal, but one of the more
+visible effects is that the Spooler component of Filebeat is removed. The
+functionality of the Spooler was similar to the one of the publisher queue from
+libbeat (the code shared by all Beats), and the presence of multiple queues
+made the performance tuning of Filebeat more complex than it needed to be.
+
+As a result, the following options are removed:
+
+- `spool_size`
+- `publish_async`
+- `idle_timeout`
+- `queue_size`
+- `bulk_queue_size`
```

Comment:
naming:
- these settings live in filebeat namespace and are used like this on reference configs: `filebeat.spool_size`, `filebeat.publish_async` and `filebeat.idle_timeout`
- these settings are 'global': `queue_size` and `bulk_max_size`.